node/sc/main_event_handler.go

```diff
@@ -80,13 +80,42 @@ func (mce *MainChainEventHandler) ConvertChildChainBlockHashToParentChainTxHash(
 	return mce.mainbridge.chainDB.ConvertChildChainBlockHashToParentChainTxHash(scBlockHash)
 }
 
-// WriteChildChainTxHash stores a transaction hash of a transaction which contains
+// decodeAndWriteChildChainTxHash decodes and stores a transaction hash of a transaction which contains
 // AnchoringData, with the key made with given child chain block hash.
 // Index is built when child chain indexing is enabled.
-func (mce *MainChainEventHandler) WriteChildChainTxHash(ccBlockHash common.Hash, ccTxHash common.Hash) {
-	mce.mainbridge.chainDB.WriteChildChainTxHash(ccBlockHash, ccTxHash)
+func (mce *MainChainEventHandler) decodeAndWriteChildChainTxHash(tx *types.Transaction) {
+	data, err := tx.AnchoredData()
+	if err != nil {
+		logger.Error("writeChildChainTxHashFromBlock : failed to get anchoring data from the tx", "txHash", tx.Hash().String())
+		return
+	}
+	anchoringData := new(types.AnchoringData)
+	if err := rlp.DecodeBytes(data, anchoringData); err != nil {
+		// Try to decode old type without a type support for compatibility.
+		anchoringDataLegacy := new(types.AnchoringDataLegacy)
+		if err := rlp.DecodeBytes(data, anchoringDataLegacy); err != nil {
+			logger.Error("writeChildChainTxHashFromBlock : failed to decode anchoring data", "txHash", tx.Hash().String())
```

Comment:
How about change error log message? 
writeChildChainTxHashFromBlock is another function, ;-; L107, L113 too

Reply:
I agree with you but we delete this method in future PR since it is not used anymore.